### PR TITLE
edit-dialog: allow user to copy spec

### PR DESCRIPTION
**Pull request by zhouhaibing089**
Fixes https://github.com/kubernetes/dashboard/issues/1870

### CODE REVIEW

File: src/app/frontend/index_module.js [hunk 1]
```diff
@@ -54,6 +54,7 @@ export default angular
     .module(
         'kubernetesDashboard',
         [
+          'angular-clipboard',
```
Comment (floreks) [inline src/app/frontend/index_module.js]: Can't this dependency be added to resource module directly? It's used only there. We should clean up our deps eventually :)

File: src/app/frontend/common/resource/editresource_controller.js [hunk 2]
```diff
@@ -67,11 +75,35 @@ export class EditResourceController {
...
    * @export
    */
   cancel() {
     this.mdDialog_.cancel();
   }
+
+  /**
+   * show an error message
+   */
+  showMessage(message) {
```
Comment (floreks) [inline src/app/frontend/common/resource/editresource_controller.js]: Our pattern for private functions is to use `_` suffix. Also please annotate this function with `@private` and `@param`.

File: src/app/frontend/common/resource/editresource_controller.js [hunk 3]
```diff
@@ -67,11 +75,35 @@ export class EditResourceController {
   }
 
   /**
+   * @export
+   */
+  copy() {
+    /* if the clipboard functionality is not supported */
+    if (!this.clipboard_.supported) {
+      this.showMessage('Your browser does not support clipboard');
```
Comment (floreks) [inline src/app/frontend/common/resource/editresource_controller.js]: All messages should be localized. You can find usage examples in other files. Look for `goog.getMsg` calls.

File: src/app/frontend/index_module.js (hunk 1 above)
Comment (zhouhaibing089) [inline src/app/frontend/index_module.js]: it should, thanks for pointing it out.

File: src/app/frontend/common/resource/editresource_controller.js (hunk 3 above)
Comment (zhouhaibing089) [inline src/app/frontend/common/resource/editresource_controller.js]: well, gotcha, I will figure it out.

File: src/app/frontend/common/resource/editresource_controller.js (hunk 2 above)
Comment (zhouhaibing089) [inline src/app/frontend/common/resource/editresource_controller.js]: sure, will do.

**Comment by zhouhaibing089** [comment]
@floreks please check it again. :)

**Comment by floreks** [comment]
LGTM.

**Pull request merged**